airflow/providers/amazon/aws/hooks/dms.py

```diff
@@ -207,7 +207,7 @@ def wait_for_task_status(self, replication_task_arn: str, status: DmsTaskWaiterS
             raise TypeError("Status must be an instance of DmsTaskWaiterStatus")
 
         dms_client = self.get_conn()
-        waiter = dms_client.get_waiter(f"replication_task_{status}")
+        waiter = dms_client.get_waiter(f"replication_task_{status.value}")
```

Comment:
This is an interesitng one - turns out that Python 3.11 changes default __str__  behaviour for Enums to return "EnumClass.ENUM" rather than the ".value". I expect this to be one of the biggest back-compatibility problem with Python 3.11 for a lot of projects.

Reply:
From https://docs.python.org/3/whatsnew/3.11.html

> Added [ReprEnum](https://docs.python.org/3/library/enum.html#enum.ReprEnum), which only modifies the [__repr__()](https://docs.python.org/3/reference/datamodel.html#object.__repr__) of members while returning their literal values (rather than names) for [__str__()](https://docs.python.org/3/reference/datamodel.html#object.__str__) and [__format__()](https://docs.python.org/3/reference/datamodel.html#object.__format__) (used by [str()](https://docs.python.org/3/library/stdtypes.html#str), [format()](https://docs.python.org/3/library/functions.html#format) and [f-string](https://docs.python.org/3/glossary.html#term-f-string)s).

> Changed [Enum.__format__()](https://docs.python.org/3/library/enum.html#enum.Enum.__format__) (the default for [format()](https://docs.python.org/3/library/functions.html#format), [str.format()](https://docs.python.org/3/library/stdtypes.html#str.format) and [f-string](https://docs.python.org/3/glossary.html#term-f-string)s) to always produce the same result as Enum.__str__(): for enums inheriting from [ReprEnum](https://docs.python.org/3/library/enum.html#enum.ReprEnum) it will be the member’s value; for all other enums it will be the enum and member name (e.g. Color.RED).